### PR TITLE
Refine README and pipeline files

### LINKED ISSUE #148: Replace `yarn` by `npm`
The `README.md` and `azure-pipelines.yml` files still mention/use `yarn` instead of `npm`.
These should consistently also use `npm`.

### PR DESCRIPTION
**Pull request by Ao-senXiong**
Fixes #148.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -30,12 +30,12 @@ steps:
 
 - bash: |
     echo ">>> Compile checker-framework-vscode"
-    yarn && yarn compile
-    # Don't have the echo below, as it masks a failure from the yarn commands.
+    npm && npm compile
```
Comment (wmdietl) [inline azure-pipelines.yml:33]: The call to `npm` without an argument doesn't seem to do the same as it did with `yarn`: https://dev.azure.com/wmdietl/checker-framework-vscode/_build/results?buildId=443&view=logs&j=43d32c17-ff19-5d0a-58b3-52a09bf40d57&t=192742a1-5cf8-5181-5ceb-6819aba4bf5f&l=13
Should those invocations be removed?

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks for working on this!

File: azure-pipelines.yml (hunk 1 above)
Comment (Ao-senXiong) [reply azure-pipelines.yml]: Yes, you are right. I just realized `npm` should use `npm run` + `scripts`.
Also, I should use `npm install` to set up the environment

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@ node_modules
 download
 .vscode
 yarn.lock
+package-lock.json
```
Comment (wmdietl) [inline .gitignore:6]: Isn't the recommendation to add this file to the repo, see e.g. https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json ?

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,4 @@ node_modules
 download
 .vscode
 yarn.lock
```
Comment (wmdietl) [inline .gitignore:5]: Should we remove this?

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks!

File: .gitignore (hunk 2 above)
Comment (Ao-senXiong) [reply .gitignore]: Thanks for the doc. I also noticed that after we added `package-lock.json`, we should use `npm ci` instead `npm install`.

**Review by wmdietl — APPROVED** [review]
Thanks!

**Pull request merged**